bundles/mapping/mapmodule/AbstractMapModule.js

```diff
@@ -551,45 +557,109 @@ Oskari.clazz.define(
+            }
+        },
+        watchUserLocation: function (successCb, errorCB, options) {
+            var me = this;
+            var sandbox = me.getSandbox();
+            var evtBuilder = Oskari.eventBuilder('UserLocationEvent');
+            var errorCodes = {1: 'denied', 2: 'unavailable', 3: 'timeout'};
+            var opts = options || {};
+            var timestamp;
+            // default values
+            var navigatorOpts = {
+                maximumAge: 5000,
+                timeout: 10000,
+                enableHighAccuracy: true
+            };
+            // override defaults with given options
+            Object.keys(navigatorOpts).forEach((key) => {
+                if (opts.hasOwnProperty(key)) {
+                    navigatorOpts[key] = opts[key];
+                }
+            });
+            if (navigator.geolocation) {
+                this._locationWatch = navigator.geolocation.watchPosition(
+                    function (position) {
+                        me.log.debug(position);
+                        if (timestamp === position.timestamp) {
+                            // some browsers sends first previous position and then changed position
+                            // TODO: is this only location emulator feature, test how this works with mobile phones
+                            return;
+                        }
+                        timestamp = position.timestamp;
+                        // transform coordinates from browser projection to current
+                        var pos = me.transformCoordinates({
+                            lon: position.coords.longitude,
+                            lat: position.coords.latitude }, 'EPSG:4326');
+                        pos.accuracy = position.coords.accuracy;
+                        sandbox.notifyAll(evtBuilder(pos.lon, pos.lat, pos.accuracy, null));
+                        // notify callback
+                        if (typeof successCb === 'function') {
+                            successCb(pos);
+                        }
+                    },
+                    function (errors) {
+                        me.log.warn('Error getting user location', errors);
+                        // notify event without position to signal failure
+                        var error = errorCodes[errors.code] || errorCodes[2];
+                        sandbox.notifyAll(evtBuilder(null, null, null, error));
+                        // me.stopUserLocationWatch();
+                        if (typeof errorCB === 'function') {
+                            errorCB(error);
+                        }
+                    }, navigatorOpts
+                );
+            } else {
+                // browser doesn't support
+            }
+        },
+        stopUserLocationWatch: function () {
+            if (navigator.geolocation) {
+                navigator.geolocation.clearWatch(this._locationWatch);
             }
```

Comment:
Do we need to have the user location tracking code in the MapModule? Could we move it somewehere else?